iceoryx_examples/waitset/README.md

```diff
@@ -1,6 +1,566 @@
+write it into a file or print it to the console. But whatever we choose to do
+we perform the same task for all the subscribers.
+
+Lets start by implementing our callback which prints the subscriber pointer, the
+payload size and the payload pointer to the console.
+```cpp
+void subscriberCallback(iox::popo::UntypedSubscriber* const subscriber)
+{
+    subscriber->take().and_then([&](iox::popo::Sample<const void>& sample) {
+        std::cout << "subscriber: " << std::hex << subscriber << " length: " << std::dec
+                  << sample.getHeader()->m_info.m_payloadSize << " ptr: " << std::hex << sample.getHeader()->payload()
+                  << std::endl;
+    });
+}
+```
+A _Trigger_ always requires a callback which has the following signature 
+`void (TriggerOrigin)`. In our example the _TriggerOrigin_ is a
+`iox::popo::UntypedSubscriber*` which we use to acquire the latest sample which the subscriber
+has received (`take()`). When `take()` was successful we print our message to
+the console inside of the `and_then` lambda.
+
+In our `main` function we create a _WaitSet_ after we registered at our central
+broker RouDi and attach our `shutdownGuard` to it to handle `CTRL+c` events. 
+```cpp
+iox::popo::WaitSet waitset;
+
+shutdownGuard.attachToWaitset(waitset);
+```
+
+After that we create a vector of 2 subscribers, subscribe and attach them to a
+_WaitSet_ with the event `HAS_NEW_SAMPLES` and the `subscriberCallback`. Everytime one 
+of the subscribers is receiving a new sample it will trigger the _WaitSet_.
+```cpp
+iox::cxx::vector<iox::popo::UntypedSubscriber, 2> subscriberVector;
+for (auto i = 0; i < subscriberVector.capacity(); ++i)
+{
+    subscriberVector.emplace_back(iox::capro::ServiceDescription{"Radar", "FrontLeft", "Counter"});
+    auto& subscriber = subscriberVector.back();
+
+    subscriber.subscribe();
+    subscriber.attachToWaitset(waitset, iox::popo::SubscriberEvent::HAS_NEW_SAMPLES, 1, subscriberCallback);
+}
+```
+
+Now our system is prepared and ready to work. We enter the event loop which 
+starts with a call to our _WaitSet_ (`waitset.wait()`). This call we block until
+one or more events triggered the _WaitSet_. After the call returned we get a
+vector filled with all the _Triggers_ which were triggered.
+
+We iterate through this vector, if a _Trigger_ originated from the `shutdownGuard`
+we exit the program otherwise we just call the assigned callback by calling
+the trigger. This will then call `subscriberCallback` with the _TriggerOrigin_
+as parameter.
+```cpp
+while (true)
+{
+    auto triggerVector = waitset.wait();
+
+    for (auto& trigger : triggerVector)
+    {
+        if (trigger.doesOriginateFrom(&shutdownGuard))
+        {
+            // CTRL+c was pressed -> exit
+            return (EXIT_SUCCESS);
+        }
+        else
+        {
+            // call the callback which was assigned to the trigger
+            trigger();
```

Comment:
Very nice, that `Trigger` are functors! Does the `UserTrigger` also have a `operator ()(..)`? What happens if I call it by accident?

Reply:
If you call `trigger()` when no callback is set then nothing happens. It is then just an empty call. `operator()(...)` is not yet defined but can maybe be an interesting feature of the future.